src/pages/index.js

```diff
@@ -1,256 +1,231 @@
-  newCardModal.close();
-}
+    .catch((err) => {
+      console.log("Failed to update avatar:", err);
+    })
+    .finally(() => {
+      avatarModalPopup.renderLoading(false);
+    });
+});
 
-// DELETE
-function handleCardDeleteSubmit(card) {
-  console.log(card._id);
-  api.deleteCard(card._id).then((message) => {
-    console.log(message);
-    card.domDeleteCard();
-    deleteConfirmModal.close();
-  });
-}
+// PREVIEW IMAGE
+const imagePopup = new PopupWithImage("#preview-image-modal");
 
-// AVATAR
-function handleAvatarChangeSubmit(Url) {
-  console.log(Url.avatarUrl);
-  api.updateAvatar(Url.avatarUrl).then((res) => {
-    console.log(res);
-  });
-  user.setAvatarPic(Url.avatarUrl);
-  submitButton.textContent = "Saving";
-  avatarChangeModal.close();
-}
+/////////////////////////////////////////////////////////   FUNCTION   ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// PREVIEW IMAGE
+function handleImageClick(name, link) {
+  imagePopup.open({ name, link });
+}
 
-// LIKE & UNLIKE FUNCTION
-function handleCardLike(cardData) {
-  if (cardData.isLiked === false) {
-    api.addLikes(cardData._id).then((res) => {
-      console.log(res);
-    });
-  } else {
-    api.removeLikes(cardData._id).then((res) => {
-      console.log(res);
+// LIKE
+function handleLikeClick(card) {
+  api
+    .setCardLikes(card.id, card.isLiked)
+    .then((newCardData) => {
+      card.updateLikes(newCardData.isLiked);
+    })
+    .catch((err) => {
+      console.log("Failed to update card likes status:", err);
     });
-    console.log(cardData._id);
-  }
 }
 
-//////////////////////////////////////////////////  ADD EVENT LISTENERS ///////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////   EVENT LISTENERS  ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-// PROFILE
+// PROFILE EDIT
 profileEditButton.addEventListener("click", () => {
-  const userInput = user.getUserInfo();
-  editFormValidator.resetValidation();
-  profileNameInput.value = userInput.name;
-  profileDescriptionInput.value = userInput.description;
-  submitButton.textContent = "Save";
-  editProfileModal.open();
+  const currentUserInfo = userInfo.getUserInfo();
+  profileTitleInput.value = currentUserInfo.name;
+  profileDescriptionInput.value = currentUserInfo.about;
+  profileEditPopup.open();
 });
 
-// CARD
-cardAddButton.addEventListener("click", () => {
-  submitButton.textContent = "Save";
-  newCardModal.open();
-  cardFormValidator.toggleButtonState();
+addCardPopup.setEventListeners();
+
+profileAddButton.addEventListener("click", () => {
+  addCardValidator.resetValidation();
```

Comment:
Method `resetValidation` is for clearing possible validation errors, when you open the **profile** form, because you insert **valid** data into its inputs. But when you add a new card, you don't have any errors after submitting. You need to call `toggleButtonState` there.